### PR TITLE
CRIU jcmd trigger checkpoint

### PR DESCRIPTION
**Pull request by JasonFengJ9**
`CRIU` `jcmd` trigger checkpoint

Enabled `jcmd` trigger a JVM checkpoint via `JDK.checkpoint` or `CRIUSupport.checkpointJVM`;
Added a test.

Co-authored with Tobi Ajila atobia@ca.ibm.com
Signed-off-by: Jason Feng <fengj@ca.ibm.com>

### CODE REVIEW

File: jcl/src/java.base/share/classes/openj9/internal/tools/attach/target/DiagnosticUtils.java [hunk 1]
```diff
@@ -78,6 +78,18 @@ public class DiagnosticUtils {
...
+/*[IF CRAC_SUPPORT]*/
+	/**
+	 * Generate a checkpoint via CRIUSupport using a compatability name.
+	 */
+	private static final String DIAGNOSTICS_JDK_CHECKPOINT = "JDK.checkpoint"; //$NON-NLS-1$
+
+	/**
+	 * Generate a checkpoint via CRIUSupport.
+	 */
+	private static final String DIAGNOSTICS_CRIUSUPPORT_CHECKPOINTJVM = "CRIUSupport.checkpointJVM"; //$NON-NLS-1$
```
Comment (tajila) [inline jcl/src/java.base/share/classes/openj9/internal/tools/attach/target/DiagnosticUtils.java:90]: We can remove this part for now

File: jcl/src/java.base/share/classes/openj9/internal/tools/attach/target/DiagnosticUtils.java [hunk 2]
```diff
@@ -485,5 +523,13 @@ private static DiagnosticProperties doHelp(String diagnosticCommand) {
 
 		commandTable.put(DIAGNOSTICS_LOAD_JVMTI_AGENT, DiagnosticUtils::loadJVMTIAgent);
 		helpTable.put(DIAGNOSTICS_LOAD_JVMTI_AGENT, DIAGNOSTICS_LOAD_JVMTI_AGENT_HELP);
+
+/*[IF CRAC_SUPPORT]*/
```
Comment (tajila) [inline jcl/src/java.base/share/classes/openj9/internal/tools/attach/target/DiagnosticUtils.java:527]: instead you can add a check here to see if Crac is enabled before adding the commands

File: jcl/src/java.base/share/classes/openj9/internal/tools/attach/target/DiagnosticUtils.java (hunk 1 above)
Comment (JasonFengJ9) [reply jcl/src/java.base/share/classes/openj9/internal/tools/attach/target/DiagnosticUtils.java:90]: Removed

File: jcl/src/java.base/share/classes/jdk/crac/Core.java [hunk 3]
```diff
@@ -63,9 +63,15 @@ public static void checkpointRestore() throws CheckpointException, RestoreExcept
...
 
 class CRIUSupportContext<R extends Resource> extends Context<R> {
-	private InternalCRIUSupport internalCRIUSupport = new InternalCRIUSupport(
-			Paths.get(InternalCRIUSupport.getCRaCCheckpointToDir())).setLeaveRunning(false).setShellJob(true)
-			.setFileLocks(true);
+	private static final String cracCheckpointDir = InternalCRIUSupport.getCRaCCheckpointToDir();
+	private final InternalCRIUSupport internalCRIUSupport = new InternalCRIUSupport(Paths.get(cracCheckpointDir))
+			.setLeaveRunning(false).setShellJob(true).setFileLocks(true);
+
+	static {
```
[the remark this answers is not in the document]
Comment (JasonFengJ9) [reply jcl/src/java.base/share/classes/jdk/crac/Core.java:70]: Removed this NPE check, and added a check `InternalCRIUSupport.isCRaCSupportEnabled()` when creating `Core.globalContext`.

File: jcl/src/java.base/share/classes/openj9/internal/tools/attach/target/DiagnosticUtils.java (hunk 2 above)
Comment (JasonFengJ9) [reply jcl/src/java.base/share/classes/openj9/internal/tools/attach/target/DiagnosticUtils.java:527]: Added the check `InternalCRIUSupport.isCRaCSupportEnabled()`.

**Review by tajila — APPROVED** [review]

**Comment by tajila** [comment]
jenkins test sanity plinux jdk21

**Pull request merged**